modules/rostests/apitests/crt/atexit.c

```diff
@@ -0,0 +1,69 @@
+/*
+* PROJECT:         ReactOS API tests
+* LICENSE:         LGPLv2.1+ - See COPYING.LIB in the top level directory
+* PURPOSE:         Test for atexit
+* PROGRAMMER:      Timo Kreuzer <timo.kreuzer@reactos.org>
+*/
+
+#include <apitest.h>
+
+#include <stdlib.h>
+#include <stdio.h>
+
+int g_sequence = 0;
+
+void exitfunc1(void)
+{
+    ok_int(g_sequence, 1);
+    g_sequence++;
+}
+
+void exitfunc2(void)
+{
+    ok_int(g_sequence, 2);
+    g_sequence++;
+}
+
+void exitfunc3(void)
+{
+    ok_int(g_sequence, 0);
```

Comment:
This checks the sequence, but there is no verification that they are _actually_ called

Reply:
Any suggestion how to fix that?

Reply:
Signalling a named event or something from the handler is probably the easiest,
You can check that it's signalled after the process exits

Reply:
I used a named sempahore to count the callbacks